### PR TITLE
feat(tt): use github as a login

### PR DESCRIPTION
**Pull request by joelhooks**
tried to make it look good, but might need some help:

Log in to Total TypeScript Total TypeScript 2023-02-17 at 09 59 51

also added it to the welcome screen as a chance to connect it.

Welcome to Total TypeScript Total TypeScript 2023-02-17 at 10 00 39

din't add it anywhere else for now

github

### CODE REVIEW

File: apps/colt-steele/package.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
     "lodash": "^4.17.21",
     "micromark": "^3.1.0",
     "next": "^13.1.2",
-    "next-auth": "^4.10.3",
+    "next-auth": "^4.19.2",
```
Comment (joelhooks) [inline apps/colt-steele/package.json:69]: updated next-auth everywhere

File: apps/total-typescript/src/components/icons.tsx [hunk 2]
```diff
@@ -19,3 +22,20 @@ export const IconGithub: React.FC<{className?: string}> = ({
...
+    />
+  </svg>
+)
```
Comment (joelhooks) [inline apps/total-typescript/src/components/icons.tsx:41]: this is dumb but the one that was already here wouldn't display and I was cutting and pasting the egghead button

File: apps/total-typescript/src/pages/api/auth/[...nextauth].ts [hunk 3]
```diff
@@ -2,23 +2,43 @@ import NextAuth, {type NextAuthOptions, Theme} from 'next-auth'
...
+    clientId: process.env.GITHUB_ID,
+    clientSecret: process.env.GITHUB_SECRET,
+    allowDangerousEmailAccountLinking: true,
+  }),
+]
+
 export const nextAuthOptions: NextAuthOptions = createOptions({
   theme: productTheme,
+  providers,
+  debug: true,
```
Comment (joelhooks) [inline apps/total-typescript/src/pages/api/auth/[...nextauth].ts:24]:
```suggestion
```

File: apps/total-typescript/src/pages/api/auth/[...nextauth].ts [hunk 4]
```diff
@@ -2,23 +2,43 @@ import NextAuth, {type NextAuthOptions, Theme} from 'next-auth'
...
   return withSentry(
-    NextAuth(req, res, createOptions({req, theme: productTheme})),
+    NextAuth(
+      req,
+      res,
+      createOptions({
+        req,
+        theme: productTheme,
+        providers,
+        debug: true,
```
Comment (joelhooks) [inline apps/total-typescript/src/pages/api/auth/[...nextauth].ts:39]:
```suggestion
```

File: apps/total-typescript/src/pages/index.tsx [hunk 5]
```diff
@@ -10,6 +10,8 @@ export const getServerSideProps: GetServerSideProps = async ({req, query}) => {
   const token = await getToken({req})
   const {products = []} = await getActiveProducts()
 
+  console.log({token})
```
Comment (joelhooks) [inline apps/total-typescript/src/pages/index.tsx:13]:
```suggestion
```

File: apps/total-typescript/src/pages/welcome/index.tsx [hunk 6]
```diff
@@ -20,12 +20,16 @@ import Image from 'next/legacy/image'
...
 import {getProduct} from 'path-to-purchase-react/products.server'
+import {IconGithub2} from '../../components/icons'
 
 export const getServerSideProps: GetServerSideProps = async ({req, query}) => {
   const {purchaseId: purchaseQueryParam, session_id, upgrade} = query
   const token = await getToken({req})
+  const providers = await getProviders()
   const {getPurchaseDetails} = getSdk()
 
+  console.log({token})
```
Comment (joelhooks) [inline apps/total-typescript/src/pages/welcome/index.tsx:31]:
```suggestion
```

File: packages/skill-api/next-auth.d.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+declare module 'next-auth' {
+  /**
+   * Returned by `useSession`, `getSession` and received as a prop on the
+   * `SessionProvider` React Context
+   */
+  interface Session {
+    role?: string
+    purchases?: any[]
+  }
```
Comment (joelhooks) [inline packages/skill-api/next-auth.d.ts:11]: you can add to the types here, which needed to happen after update to next-auth

**Comment by vojtaholik** [comment]
@joelhooks I got the styles updated. 

shall we add new github oath app for production? I noticed the current one is pointing at localhost and has a "dev" suffix. 

how did you test it? it keeps pointing at production url for me and I couldn't figure out where that's coming from.

also yay for having an ability to display users profile pics. :)

**Comment by joelhooks** [comment]
@vojtaholik the prod github app is already setup. important to use Github App and NOT Oauth App. I think the keys for dev are in vercel? if not they are in github org

**Pull request merged**